qiskit_experiments/calibration/calibrations.py

```diff
@@ -0,0 +1,912 @@
+        careful extraction of the qubits from the subroutine and in the appropriate
+        order. Next, the parameters are identified and assigned. This is needed to
+        handle situations where the same parameterized schedule is called but on
+        different channels. For example,
+
+        .. code-block:: python
+
+            ch0 = Parameter("ch0")
+            ch1 = Parameter("ch1")
+
+            with pulse.build(name="xp") as xp:
+                pulse.play(Gaussian(duration, amp, sigma), DriveChannel(ch0))
+
+            with pulse.build(name="xt_xp") as xt:
+                pulse.call(xp)
+                pulse.call(xp, value_dict={ch0: ch1})
+
+        Here, we define the xp schedule for all qubits as a Gaussian. Next, we define a
+        schedule where both xp schedules are called simultaneously on different channels.
+
+        Args:
+            schedule: The schedule with assigned channel indices for which we wish to
+                assign values to non-channel parameters.
+            qubits: The qubits for which to get the schedule.
+            free_params: The parameters that are to be left free. See get_schedules for details.
+            group: The calibration group of the parameters.
+            cutoff_date: Retrieve the most recent parameter up until the cutoff date. Parameters
+                generated after the cutoff date will be ignored. If the cutoff_date is None then
+                all parameters are considered. This allows users to discard more recent values that
+                may be erroneous.
+
+        Returns:
+            ret_schedule: The schedule with assigned parameters.
+
+        Raises:
+            CalibrationError:
+                - If a channel has not been assigned.
+                - If a parameter that is needed does not have a value.
+        """
+        # 1) Restrict the given qubits to those in the given schedule.
+        qubit_set = set()
+        for chan in schedule.channels:
+            if isinstance(chan.index, ParameterExpression):
+                raise (
+                    CalibrationError(
+                        f"All parametric channels must be assigned before searching for "
+                        f"non-channel parameters. {chan} is parametric."
+                    )
+                )
+            if isinstance(chan, (DriveChannel, MeasureChannel)):
+                qubit_set.add(chan.index)
+
+            if isinstance(chan, ControlChannel):
+                for qubit in self._controls_config_r[chan]:
+                    qubit_set.add(qubit)
+
+        qubits_ = tuple(qubit for qubit in qubits if qubit in qubit_set)
+
+        # 2) Recursively assign the parameters in the called instructions.
+        ret_schedule = ScheduleBlock(
+            alignment_context=schedule.alignment_context,
+            name=schedule.name,
+            metadata=schedule.metadata,
+        )
+
+        for inst in schedule.blocks:
+            if isinstance(inst, Call):
```

Comment:
Perhaps the `Schedule` here is just an instruction (not a program), so we don't need to manage it on our end. Perhaps serialization would be the problem, but this should be supported by terra. I'm thinking the best way to serialize the `Schedule` and `ScheduleBlock` now. Once we decide this format, we cannot change it again because serialized data should be stable and not sensitive to code version change.

Reply:
Anyways I think it make sense to remove schedule subroutine. It makes the code and test more simple, and perhaps we can avoid some edge cases. The only place currently we get the schedule is the instruction schedule map, since builder output is completely replaced with block. We don't need to assume we call subroutine from instmap.